examples/loader-duckdb/src/index.md

```diff
@@ -0,0 +1,157 @@
+# DuckDB data loader
+
+Here’s a bash data loader that uses curl to download a CSV file from Eurostat (the data portal from the statistical office of the European Union), then calls [DuckDB](https://duckdb.org/) to filter it and saves the result as [Apache Parquet](https://observablehq.com/framework/lib/arrow#apache-parquet).
+
+The example illustrates the technique with statistics about education to modern foreign languages across Europe ([educ_uoe_lang01](https://ec.europa.eu/eurostat/databrowser/view/educ_uoe_lang01/default/table?lang=en&category=educ.educ_lang.educ_uoe_lang)).
+
+The data loader lives in [`src/educ_uoe_lang01.parquet.sh`](./src/educ_uoe_lang01.parquet.sh).
+
+Because the Eurostat API does not compress data as part of the http response, but instead offers to send compressed data, we download it outside of DuckDB — here using [curl](https://curl.se/) — then uncompress it with [gunzip](https://en.wikipedia.org/wiki/Gzip). The raw file is saved to a temporary directory, which makes it much faster to iterate on the data loader when we want to tweak the query, for example to change the education level or the time period considered.
```

Comment:
```suggestion
Because the Eurostat API does not compress data as part of the http response, but instead offers to send compressed data, we download it outside of DuckDB — here using [curl](https://curl.se/) — then uncompress it with [gunzip](https://en.wikipedia.org/wiki/Gzip). The raw file is saved to a temporary directory, which makes it much faster to iterate on the data loader when we want to tweak the query, for example to change the education level or the time period considered.
```

My impression is that we’re downloading it outside of DuckDB mostly so that we can cache the downloaded file, rather than downloading it each time we run the data loader? Though the risk of that approach is that we must remember to clear a separate cache directory in `$TMPDIR` rather than the normal one in `.observablehq/cache`. So, can we use the latter instead, so that clearing the cache will re-download the file? I would think that using the existing cache would be a best practice that we should recommend.

Reply:
Unfortunately the Eurostat API does not send "csv on an http gziped connection", but instead sends a gzip file ("compressed csv"). I haven't managed to access it directly with DuckDB.

Reply:
I've adopted `.observablehq/cache`; I'm using a subfolder though, to separate the data loader output from its framework copy (even though it works locally if we cat from a file to stdout, then let framework overwrite the file… it felt dangerous).